compose/config/config.py

```diff
@@ -493,6 +517,22 @@ def parse_environment(environment):
     )
 
 
+def parse_build_arguments(arguments):
+    if not arguments:
+        return {}
+
+    if isinstance(arguments, list):
+        return dict(split_env(e) for e in arguments)
+
+    if isinstance(arguments, dict):
+        return dict(arguments)
+
+    raise ConfigurationError(
+        "build arguments \"%s\" must be a list or mapping," %
+        arguments
+    )
```

Comment:
Since this is almost identical to `parse_environment()` (and really `parse_labels()` is as well), I think we should have just one function that takes two params for the differences between them:

```python

def parse_dict_or_list(split_func, type_name, arguments):
    ...

parse_build_arguments = functools.partial(parse_dict_or_list, split_env, 'build arguments')
parse_environment = functools.partial(parse_dict_or_list, split_env, 'build arguments')
parse_labels = functools.partial(parse_dict_or_list, split_label, 'labels')
```

These could be single line functions instead of using `functools.partial`, I'm *partial* to functools.